connmgr/connmanager.go

```diff
@@ -288,7 +292,30 @@ func (cm *ConnManager) NewConnReq() {
 		cm.requests <- handleFailed{c, err}
 		return
 	}
-	c.Addr = addr
+
+	host, strPort, err := net.SplitHostPort(addr)
```

Comment:
Since everything else in the package now deals with `net.Addr`, I think this should be modified instead such that the `GetNewAddress` callback returns a `net.Addr` and then the `newAddressFunc` closure in `server.go` which is provided would pick up the parsing, lookup, and `net.Addr` creation.  This would work as is, but it seems to go against the spirit of the PR which seems to be aimed at making it agnostic to the specific address type.  In fact, I believe that would get rid of the need to even pass in a lookup function at all and allow the code you added in `server.go` under the `connectNodeMsg` case and `newServer` function to be refactored into a common code used in all three places (`connectNodeMsg`, `newServer`, and `newAddressFunc`).  What do you think?

I expect `addrmgr.GetAddress` itself should probably also be modified at some point to return `net.Addr` too, but that can be saved for another PR.